### ISSUE TITLE
Error: Invalid count argument

### ISSUE THREAD

**Issue by kkalinowski2021** [terraform-google-modules/terraform-google-kubernetes-engine#1087]
### TL;DR

Hello,
I am new here. I just started configuring the GKE module, declared all necessary variables, run terraform plan all good, however when I run terraform apply. I get this message.

Error: Invalid count argument

  on .terraform/modules/gke.gcloud_delete_default_kube_dns_configmap/main.tf line 63, in resource "null_resource" "module_depends_on":
  63:   count = length(var.module_depends_on) > 0 ? 1 : 0

The "count" value depends on resource attributes that cannot be determined
until apply, so Terraform cannot predict how many instances will be created.
To work around this, use the -target argument to first apply only the
resources that the count depends on.

My source is source = "terraform-google-modules/kubernetes-engine/google"
I googled this message but the only suggestion was to delete state file which I did and I am getting the same error.
Any help would be greatly appreciated.
Thanks,

### Terraform Configuration

```hcl
# insert modules reqs and provixder reqs.
provider "google" {
  region  = var.region
  project = var.gcp_project

}

provider "google-beta" {
  project = var.gcp_project
  region  = var.region

}

resource "google_storage_bucket" "k8stfstate2" {
  name     = "trip-${var.environment}-k8stfstate2"
  location = upper(var.region)

  versioning {
    enabled = true
  }

  labels = var.labels
}

terraform {

  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "~> 3.90.0"
    }
    google-beta = {
      source  = "hashicorp/google-beta"
      version = "~> 3.90.0"
    }
    random = {
      source = "hashicorp/random"
      version = "~> 3.1.0"
    }
  }

  required_version = "~> 0.13"
}
```

### Terraform Version

```sh
Terraform v0.13.7
```

**Comment by morgante** [comment]
Please share the config where you actually call the GKE module.

**Comment by kkalinowski2021** [comment]
I have cluster.tf file 

# google_client_config and kubernetes provider must be explicitly specified like the following.
data "google_client_config" "default" {}

provider "kubernetes" {
  host                   = "https://${module.gke.endpoint}"
  token                  = data.google_client_config.default.access_token
  cluster_ca_certificate = base64decode(module.gke.ca_certificate)
}

module "gke" {
  source                     = "terraform-google-modules/kubernetes-engine/google"
  project_id                 = var.gcp_project
  name                       = var.name
  region                     = var.region
  zones                      = var.zones
  network                    = var.gcp_vpc_network
  subnetwork                 = var.gcp_vpc_subnet
  ip_range_pods              = var.gcp_ip_range_pods
  ip_range_services          = var.gcp_ip_range_services
  http_load_balancing        = false # is this https/ http ?
  horizontal_pod_autoscaling = true
  network_policy             = true

  node_pools = [
    {
      name               = "blue"
      auto_repair        = true
      auto_upgrade       = true
      disk_size_gb       = 100
      disk_type          = "pd-standard"
      image_type         = "COS"
      initial_node_count = 5
      local_ssd_count    = 0
      machine_type       = "n1-standard-8"
      max_count          = 10
      min_count          = 2
      node_locations     = join(", ", var.zones)
      preemptible        = false
      #service_account    = "project-service-account@${var.gcp_project}.iam.gserviceaccount.com"
    },
    {
      name               = "green"
      auto_repair        = true
      auto_upgrade       = true
      disk_size_gb       = 100
      disk_type          = "pd-standard"
      image_type         = "COS"
      initial_node_count = 5
      local_ssd_count    = 0
      machine_type       = "n1-standard-8"
      max_count          = 10
      min_count          = 2
      node_locations     = join(", ", var.zones)
      preemptible        = false
      #service_account    = "project-service-account@${var.gcp_project}.iam.gserviceaccount.com"
    },
    {
      name               = "preemptible"
      auto_repair        = true
      auto_upgrade       = true
      disk_size_gb       = 100
      disk_type          = "pd-standard"
      image_type         = "COS"
      initial_node_count = 5
      local_ssd_count    = 0
      machine_type       = "n1-standard-8"
      max_count          = 10
      min_count          = 2
      node_locations     = join(", ", var.zones)
      preemptible        = true
      #service_account    = "project-service-account@${var.gcp_project}.iam.gserviceaccount.com"
    }
  ]

  node_pools_oauth_scopes = {
    all = []

    default-node-pool = [
      "https://www.googleapis.com/auth/cloud-platform",
    ]
  }

  node_pools_labels = {
    all = {}

    default-node-pool = {
      default-node-pool = true
    }
  }

  node_pools_metadata = {
    all = {}

    default-node-pool = {
      node-pool-metadata-custom-value = "my-node-pool"
    }
  }

  node_pools_taints = {
    all = []

    default-node-pool = [
      {
        key    = "default-node-pool"
        value  = true
        effect = "PREFER_NO_SCHEDULE"
      },
    ]
  }

  node_pools_tags = {
    all = []

    default-node-pool = [
      "default-node-pool",
    ]
  }
}

**Comment by markmacmahon** [comment]
I came across the same issue in the terraform-example-foundation-app:
rather than deleting the state, i found it more helpful to delete the node_pools
1. Get list of node_pools from terraform state `terraform state list | grep node_pools´
2. Destroy each node pool from above `terraform destroy -target  '$NODE_POOL'
3. Terraform plan

**Comment by kkalinowski2021** [comment]
HI Mark,
Thanks for your suggestion but this did not work for me :( however I was able to deploy the gke using latest terraform version 1.1.2. Terraform v0.13.7 is not compatible.
What version of the terraform did you use?
